django/db/backends/base/schema.py

```diff
@@ -121,6 +121,15 @@ def __exit__(self, exc_type, exc_value, traceback):
 
     # Core utility functions
 
+    def prepare_param(self, node):
+        sql, params = None, (node,)
+        if hasattr(node, 'as_sql'):
+            compiler = self.connection.ops.compiler('SQLCompiler')(
+                query=None, connection=self.connection, using=None
+            )
+            sql, params = compiler.compile(node)
+        return sql, tuple(params)
```

Comment:
I think it's a bit neater to invert the check for `node.as_sql` and return early:

```suggestion
        if not hasattr(node, 'as_sql'):
            return None, (node,)
        compiler = self.connection.ops.compiler('SQLCompiler')(
            query=None, connection=self.connection, using=None
        )
        sql, params = compiler.compile(node)
        return sql, tuple(params)
```

Reply:
hm… this will increase mccabe complexity and I also personally don't think a negative if statement will increase readability. If you don't mind, I will keep it as i.s

Reply:
I agree with @Ian-Foote. For example, the first line isn't used in the case that `hasattr(node, 'as_sql')` is true.